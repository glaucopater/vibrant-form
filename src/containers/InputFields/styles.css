.vibrantFormInputFields {
    display: inline-grid;
    grid-auto-flow: column;
    grid-template-rows: repeat(5, auto);
    grid-gap: 8px;
    width: 100%;
}

@media only screen and (max-width: 600px) {
    .vibrantFormInputFields {
        display: flex;
        flex-direction: column;
    }
}