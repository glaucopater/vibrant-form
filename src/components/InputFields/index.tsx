
import React from "react";
import Input from "../../components/Input";
import { FormPropsType } from "../../types";
import "./styles.css";

const InputFields: React.FC<FormPropsType> = ({ fieldsData }) => {
    const content = fieldsData && fieldsData.map((field, index) =>
        <Input
            key={index.toString()}
            {...field}
        />);
    return <div className="vibrantFormInputFields">{content}</div>;
}

export default InputFields;