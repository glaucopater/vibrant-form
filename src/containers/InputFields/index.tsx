
import React from "react";
import Input from "../../components/Input";
import { FormPropsType } from "../../types";
import "./styles.css";

const InputFields: React.FC<FormPropsType> = ({ fieldsData, formErrors, component: Component = Input, withTheme }) => {
    const isValidated = typeof formErrors !== "undefined";
    const content = fieldsData && fieldsData.map((field, index) =>
        <Component
            key={index.toString()}
            {...field}
            isValidated={isValidated}
            withTheme={withTheme}
            errors={formErrors ? formErrors?.filter(error => error.name === field.name) : []
            }
        />);

    return <div className={withTheme ? "vibrantFormInputFields" : ""}>{content}</div>;
}

export default InputFields;