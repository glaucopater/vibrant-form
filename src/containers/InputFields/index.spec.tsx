import InputFields from ".";

describe("InputFields", () => {
    it("should render the empty component", () => {
        expect(InputFields).toMatchSnapshot();
    })
})
