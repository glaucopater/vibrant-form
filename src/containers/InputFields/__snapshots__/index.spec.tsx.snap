// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputFields should render the empty component 1`] = `[Function]`;
